sql/moz-fx-data-shared-prod/monitoring_derived/bigquery_usage_v2/query.py

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env python3
+
+"""
+Determine BigQuery usage.
+
+To read more on the source table, please visit:
+https://cloud.google.com/bigquery/docs/information-schema-jobs-by-organization
+"""
+
+from argparse import ArgumentParser
+
+from google.cloud import bigquery
+
+DEFAULT_PROJECTS = [
+    "moz-fx-data-shared-prod",
+]
+
+parser = ArgumentParser(description=__doc__)
+parser.add_argument("--date", required=True)  # expect string with format yyyy-mm-dd
+parser.add_argument("--project", default="moz-fx-data-shared-prod")
+parser.add_argument("--source_projects", nargs="+", default=DEFAULT_PROJECTS)
+parser.add_argument("--destination_dataset", default="monitoring_derived")
+parser.add_argument("--destination_table", default="bigquery_usage_v2")
+
+
+def create_query(date, source_project):
+    """Create query for a source project."""
+    return f"""
+        SELECT
+         "{source_project}" AS source_project,
+          DATE('{date}') AS creation_date,
+          job_id,
+          job_type,
+          reservation_id,
+          cache_hit,
+          state,
+          statement_type,
+          project_id AS project_id,
+          referenced_tables.project_id AS reference_project_id,
+          dataset_id AS reference_dataset_id,
+          table_id AS reference_table_id,
+          destination_table.project_id AS  destination_project_id,
+          destination_table.dataset_id AS  destination_dataset_id,
+          destination_table.table_id AS  destination_table_id,
+          user_email,
+          end_time-start_time as task_duration,
+          ROUND(total_bytes_processed / 1024 / 1024 / 1024 / 1024, 4)
+          AS total_terabytes_processed,
+          total_slot_ms,
+          error_result.location AS error_location,
+          error_result.reason AS error_reason,
+          error_result.message AS error_message,
+          query_info.resource_warning AS resource_warning,
+        FROM
+          `{source_project}.region-us.INFORMATION_SCHEMA.JOBS_BY_ORGANIZATION`
```

Comment:
The main issue here is that `source_project` in this context only specifies where this specific query is to be run. It doesn't do any kind of filtering on jobs to a specific project. So the output of this query will return metadata for jobs that have no interaction with the data platform, when our intent is to limit the metadata to only jobs interacting with the platform.

In order to limit metadata to jobs reading from or writing to destinations listed in `DEFAULT_PROJECTS`, I think you need to perform some kind of filtering in the query itself. So instead of iterating over `DEFAULT_PROJECTS` one at a time to get `JOBS_BY_PROJECT` for each, you should query `JOBS_BY_ORGANIZATION` once where `reference_project_id` and/or `destination_project_id`  are in `DEFAULT_PROJECTS`. `source_project` in this case, instead of being static, would be what is currently the value of `project_id`: the project the job ran in.

Reply:
Thanks for sharing the details.  They are very helpful.  I have added the filters for `DEFAULT PROJECTS` in the query.  I also needed the jobs that run under the defaults projects.  I have been getting this data from the `JOBS_BY_PROJECT` view in the `bigquery_usage_v1` table.  I hope that works.